node/header_exchange.go

```diff
@@ -213,7 +216,11 @@ func newP2PExchange(
 	conngater *conngater.BasicConnectionGater,
 	opts ...goheaderp2p.Option[goheaderp2p.ClientParameters],
 ) (*goheaderp2p.Exchange[*types.SignedHeader], error) {
-	return goheaderp2p.NewExchange[*types.SignedHeader](host, peers, network, conngater, opts...)
+	opts = append(opts,
+		goheaderp2p.WithNetworkID[goheaderp2p.ClientParameters](network),
+		goheaderp2p.WithChainID[goheaderp2p.ClientParameters](network),
```

Comment:
should you use `hExService.genesis.ChainID` for `WithChainID`? or `network` is fine?

Reply:
The given WithChainID must match the ChainID in the headers. The given WithNetworkID can be anything but must be equal to the peers in the same network. We conceptually split those into two things, but they may be the same and usually are. 

The `network` passed to the constructor is the chain id from genesis doc, as per the code, so we can use it here. It will also be equal for each node, so we can also set network id unless you have a different notion of the network, which I am not aware of.

Reply:
agree, however if we start differentiating (which i think we should, otherwise why two things representing the same and also update the code), this can create unnecessary bug. why not `hExService.genesis.ChainID` to avoid future problems?

Reply:
We don't have access to this field in the method, but can add another param

Reply:
sure. let's have chainId passed to it. thanks